sotodlib/io/g3tsmurf_utils.py

```diff
@@ -129,13 +138,41 @@ def get_batch(
     """
     
     session = archive.Session()
-    obs = session.query(Observations).filter(Observations.obs_id==obs_id).one()
+
+    if isinstance(obs_id, Observations):
+        obs = obs_id
+        obs_id = obs.obs_id
+    else:
+        obs = session.query(Observations).filter(Observations.obs_id==obs_id).one()
     db_files = session.query(Files).filter(Files.obs_id==obs_id).order_by(Files.start)
     filenames = sorted( [f.name for f in db_files])
     
     ts = obs.tunesets[0] ## if this throws an error we have some fallbacks
     obs_dets, obs_samps = len(ts.dets), obs.n_samples
     session.close()
+
+    ## should we let folks overwrite this here?
+    if "archive" in load_file_args:
+        archive = load_file_args.pop("archive")
+        
+    if "status" in load_file_args:
+        status = load_file_args.pop("status")
+    else:
+        status = SmurfStatus.from_file(filenames[0])
+
+    if start == None:
+        start = obs.start
+    if end == None:
+        end = obe.stop
+    samprate = 4e3 / (status.downsample_enabled * status.downsample_factor)
```

Comment:
Doesn't this divide by 0 if `status.downsample_enabled` is `False`?

Reply:
Yeah, that's definitely what it does.  Will fix

Reply:
I expect `status.downsample_factor` is 1 when it isn't downsampled. But I may be wrong.